x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts

```diff
@@ -289,3 +293,47 @@ export const convertToSnakeCase = <T extends Record<string, unknown>>(
     return { ...acc, [newKey]: obj[item] };
   }, {});
 };
+
+export const mergeStatuses = (
+  id: string,
+  failures: Array<SavedObjectsFindResult<IRuleStatusAttributes>>,
+  acc: RuleStatusResponse
+) => {
+  if (failures.length === 0) {
+    return {
+      ...acc,
+    };
+  }
+  const convertedCurrentStatus = convertToSnakeCase<IRuleStatusAttributes>(failures[0].attributes);
+  return {
+    ...acc,
+    [id]: {
+      current_status: convertedCurrentStatus,
+      failures: failures.map((errorItem) =>
+        convertToSnakeCase<IRuleStatusAttributes>(errorItem.attributes)
+      ),
+    },
+  } as RuleStatusResponse;
+};
+
+export const getFailingRules = (ids: string[], alertsClient: AlertsClient) =>
+  Promise.all(
```

Comment:
I think we need to handle the potential rejections/errors thrown by the alertsClient, here. Might be easier to convert to a try/catch/await.